lib/preload.js

```diff
@@ -149,35 +147,32 @@ const runPreloadContainerAndPull = (docker, container, imageId, dockerPort) => {
   }
   const removeListener = () => {
     // Set things back to normal.
-    if (process.stdin.setRawMode) {
-      process.stdin.setRawMode(false)
-      process.stdin.pause()
-      process.stdin.removeListener('data', killContainer)
+    if (stdin && stdin.setRawMode) {
+      stdin.setRawMode(false)
+      stdin.pause()
+      stdin.removeListener('data', killContainer)
```

Comment:
I'm wondering if it'd make more sense to handle the Ctrl+C stuff in `bin/resin-preload` and the resin-cli respectively, as this is something consumers should (or would want) to handle themselves...?

Reply:
I'm not sure what would be the best way to do this.
Should we add an argument to `preload.run()` ? An EventEmitter so the caller could call `canceller.emit('cancel')` ?

Reply:
Or maybe it should be returned by `run` instead of being an argument?
So it would return `{ statusCodePromise, output, fetchSizeProgress, dockerPullProgress, canceller }`

Reply:
@jhermsmeier I updated this PR, now all functions that start a docker container return a cancel function.
That way we can intercept ctrl+c, kill the container, remove it and exit instead of leaving unused docker containers.